### ISSUE TITLE
Build failure: `python3Packages.bidict` on darwin

### ISSUE THREAD

**Issue by shivaraj-bh** [NixOS/nixpkgs#299933]
### Steps To Reproduce

1. `nix build .#python3Packages.bidict`

### Build log

hydra build: https://hydra.nixos.org/build/254324381
build logs: https://cache.nixos.org/log/wb1390fl6hpf5sg5pzb2l071mflrlkpc-python3.11-bidict-0.23.1.drv

### Notify maintainers

@jab @jbaum98

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.2.0, macOS 14.2.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/nix/store/bv68i4l7g6s5sixxm0c9lcjdxhlkg0n5-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by jab** [comment]
Not sure why this suddenly started failing after it worked fine for the last 11 days.

Reproduced this and submitted a patch in #300044 that fixed it for me locally.

**Comment by shivaraj-bh** [comment]
> Reproduced this and submitted a patch in https://github.com/NixOS/nixpkgs/pull/300044 that fixed it for me locally.

Thanks, I can confirm that the patch does work.

**Comment by shivaraj-bh** [comment]
> Not sure why this suddenly started failing after it worked fine for the last 11 days.

In my observation, it only breaks on [this commit](10bca7eddf27daf4206899bedf36b410389de0a4), the [commit before that](76ea2eb90fc9f732662055fc0e07b3e155ad2e84) works just fine. That's extremely weird.

**Comment by shivaraj-bh** [comment]
I logged the test phase on the latest commit and the one before that, here’s what’s changed:

- Before
```sh
platform darwin -- Python 3.11.8, pytest-7.4.4, pluggy-1.4.0
rootdir: /dev
configfile: null
plugins: hypothesis-6.91.0, xdist-3.5.0
```

- After

```sh
platform darwin -- Python 3.11.8, pytest-8.0.2, pluggy-1.4.0
rootdir: /dev
configfile: null
plugins: hypothesis-6.98.17, xdist-3.5.0
```

Looks like it is to do with the [breaking changes introduced in pytest 8.0](https://docs.pytest.org/en/stable/changelog.html#pytest-8-0-0rc1-2023-12-30)

**Comment by jab** [comment]
Nice work, @shivaraj-bh. 🕵️

It looks like 92c34d01035c9bca16ad8d5b6c2babdad24bb87a is the commit that upgraded to pytest 8, but that was [merged](#300044) two weeks ago. So I'm not sure how this passed for https://github.com/NixOS/nixpkgs/commit/76ea2eb90fc9f732662055fc0e07b3e155ad2e84 4 days ago.

**Comment by shivaraj-bh** [comment]
The build that passed 4 days ago used `pytest.ini` as `ConfigFile` and the `rootdir` also wasn’t pointing to `/dev`, here’s a snippet from the logs:
```sh
platform darwin -- Python 3.11.8, pytest-8.0.2, pluggy-1.4.0 -- /nix/store/c05vbvkjxarxkws9zkwrcwrzlsx9nd68-python3-3.11.8/bin/python3.11
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase(PosixPath('/private/tmp/nix-build-python3.11-bidict-0.23.1.drv-0/source/.hypothesis/examples'))
benchmark: 4.0.0 (defaults: timer=time.perf_counter disable_gc=True min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /private/tmp/nix-build-python3.11-bidict-0.23.1.drv-0/source
configfile: pytest.ini
testpaths: bidict, tests, docs
plugins: hypothesis-6.98.17, xdist-3.5.0, benchmark-4.0.0
```

The log file: https://cache.nixos.org/log/y8yl83syh1wdm6y6sigdw03pz4m78mvz-python3.11-bidict-0.23.1.drv

But the latest build which failed, had `rootDir` set to `/dev` and `configFile` set to `null`.
